idl.cmake

```diff
@@ -51,21 +51,81 @@ MACRO(GENERATE_IDL_FILE FILENAME DIRECTORY)
   IF(${OMNIIDL} STREQUAL OMNIIDL-NOTFOUND)
     MESSAGE(FATAL_ERROR "cannot find omniidl.")
   ENDIF(${OMNIIDL} STREQUAL OMNIIDL-NOTFOUND)
+
+  LIST(APPEND IDL_COMPILED_FILES ${FILENAME}SK.cc ${FILENAME}.hh)
   ADD_CUSTOM_COMMAND(
-    OUTPUT ${FILENAME}SK.cc ${FILENAME}.hh
+    OUTPUT ${IDL_COMPILED_FILES}
     COMMAND ${OMNIIDL}
     ARGS -bcxx -Wbkeep_inc_path ${_OMNIIDL_INCLUDE_FLAG}
     -C${_PATH} ${DIRECTORY}/${_NAME}.idl
     MAIN_DEPENDENCY ${DIRECTORY}/${_NAME}.idl
+    COMMENT "Compiling C++ stubs for ${_NAME}"
     )
-  SET(ALL_IDL_STUBS ${FILENAME}SK.cc ${FILENAME}.hh ${ALL_IDL_STUBS})
+
+  SET(ALL_IDL_CPP_STUBS ${IDL_COMPILED_FILES} ${ALL_IDL_CPP_STUBS})
 
   # Clean generated files.
   SET_PROPERTY(
     DIRECTORY APPEND PROPERTY
     ADDITIONAL_MAKE_CLEAN_FILES
-    ${FILENAME}SK.cc ${FILENAME}.hh
+    ${IDL_COMPILED_FILES}
     )
 
-  LIST(APPEND LOGGING_WATCHED_VARIABLES OMNIIDL ALL_IDL_STUBS)
+  LIST(APPEND LOGGING_WATCHED_VARIABLES OMNIIDL ALL_IDL_CPP_STUBS)
+ENDMACRO(GENERATE_IDL_CPP FILENAME DIRECTORY)
+
+# GENERATE_IDL_PYTHON FILENAME DIRECTORY
+# ------------------------------------
+#
+# Generate Python stubs from an idl file.
+# An include directory can also be specified.
+#
+# FILENAME : IDL filename without the extension
+#            Can be prefixed by a path: _path/_filename
+# DIRECTORY : IDL directory
+#             The idl file being search for is: ${DIRECTORY}/${_filename}.idl
+#
+MACRO(GENERATE_IDL_PYTHON FILENAME DIRECTORY)
+  GET_FILENAME_COMPONENT (_PATH ${FILENAME} PATH)
+  GET_FILENAME_COMPONENT (_NAME ${FILENAME} NAME)
+  IF (_PATH STREQUAL "")
+    SET(_PATH "./")
+  ENDIF (_PATH STREQUAL "")
+  FIND_PROGRAM(OMNIIDL omniidl)
+  IF(${OMNIIDL} STREQUAL OMNIIDL-NOTFOUND)
+    MESSAGE(FATAL_ERROR "cannot find omniidl.")
+  ENDIF(${OMNIIDL} STREQUAL OMNIIDL-NOTFOUND)
+  ADD_CUSTOM_COMMAND(
+    OUTPUT ${FILENAME}_idl.py
+    COMMAND ${OMNIIDL}
+    ARGS -bpython ${_OMNIIDL_INCLUDE_FLAG}
+    -C${_PATH} ${DIRECTORY}/${_NAME}.idl
+    MAIN_DEPENDENCY ${DIRECTORY}/${_NAME}.idl
+    COMMENT "Compiling Python stubs for ${_NAME}"
+    )
+  SET(ALL_IDL_PYTHON_STUBS ${FILENAME}_idl.py ${ALL_IDL_PYTHON_STUBS})
+
+  # Clean generated files.
+  SET_PROPERTY(
+    DIRECTORY APPEND PROPERTY
+    ADDITIONAL_MAKE_CLEAN_FILES
+    ${FILENAME}_idl.py
+    )
+
+  LIST(APPEND LOGGING_WATCHED_VARIABLES OMNIIDL ALL_IDL_PYTHON_STUBS)
+ENDMACRO(GENERATE_IDL_PYTHON FILENAME DIRECTORY)
+
+# GENERATE_IDL_FILE FILENAME DIRECTORY
+# ------------------------------------
+#
+# Generate C++ stubs from an idl file (legacy macro).
+# An include directory can also be specified.
+#
+# FILENAME : IDL filename without the extension
+#            Can be prefixed by a path: _path/_filename
+# DIRECTORY : IDL directory
+#             The idl file being search for is: ${DIRECTORY}/${_filename}.idl
+#
+MACRO(GENERATE_IDL_FILE FILENAME DIRECTORY)
```

Comment:
IMHO, we should get rid of this macro, I don't think backward compatibility is very important for this feature:
a\ It's used by very few projects (actually the only one that comes to mind now is for the integration between the pre-python stack of tasks and the old version of OpenHRP)
b\ to be fully retro-compatible, we would need to also reproduce the variables (i.e. `ALL_IDL_STUBS`)
c\ the name of the macro was confusing in the first place